packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/LightCurveViewer/LightCurveViewerContainer.spec.js

```diff
@@ -1,20 +1,187 @@
 import { shallow } from 'enzyme'
 import React from 'react'
+import nock from 'nock'
+import sinon from 'sinon'
+import * as d3 from 'd3'
+import { zip } from 'lodash'
+import asyncStates from '@zooniverse/async-states'
 
 import LightCurveViewerContainer from './LightCurveViewerContainer'
+import mockData from './mockData.json'
+import { Factory } from 'rosie'
 
 let wrapper
 
+const subject = Factory.build('subject', { locations: [
+  { 'application/json': 'http://localhost:8080/mockData.json' }
+]})
+
+const nextSubject = Factory.build('subject', { locations: [
+  { 'text/plain': 'http://localhost:8080/nextSubject.json' }
+] })
+
+const nextSubjectJSON = { x: [1, 2], y: [3, 4] }
+
+const imageSubject = Factory.build('subject')
+
+const failSubject = Factory.build('subject', {
+  locations: [
+    { 'application/json': 'http://localhost:8080/failure.json' }
+  ]
+})
+
 describe('Component > LightCurveViewerContainer', function () {
-  beforeEach(function () {
+  it('should render without crashing', function () {
     wrapper = shallow(<LightCurveViewerContainer.wrappedComponent />)
+    expect(wrapper).to.be.ok
   })
 
-  it('should render without crashing', function () {
-    expect(wrapper).to.be.ok
+  it('should mount with an initialized state', function () {
+    wrapper = shallow(<LightCurveViewerContainer.wrappedComponent />, { disableLifecycleMethods: true })
+    const mockState = {
+      dataExtent: { x: [], y: [] },
+      dataPoints: []
+    }
+    expect(wrapper.state()).to.eql(mockState)
+  })
+
+  describe('without a subject', function () {
+    it('should render null with the default props', function () {
+      wrapper = shallow(<LightCurveViewerContainer.wrappedComponent />)
+      expect(wrapper.html()).to.be.null
+    })
+  })
+
+  describe('with an invalid subject', function () {
+    let cdmSpy
+    let onErrorSpy
+    let nockScope
+    before(function () {
+      sinon.stub(console, 'error')
+      cdmSpy = sinon.spy(LightCurveViewerContainer.wrappedComponent.prototype, 'componentDidMount')
+      onErrorSpy = sinon.spy()
+      nockScope = nock('http://localhost:8080')
+        .persist(true)
+        .get('/failure.json')
+        .reply(404)
+    })
+
+    afterEach(function () {
+      cdmSpy.resetHistory()
+      onErrorSpy.resetHistory()
+    })
+
+    after(function () {
+      console.error.restore()
+      cdmSpy.restore()
+      nock.cleanAll()
+      nockScope.persist(false)
+    })
+
+    it('should error if a json or text subject location file can\'t be found', function (done) {
+      wrapper = shallow(
+        <LightCurveViewerContainer.wrappedComponent onError={onErrorSpy} subject={imageSubject} />
+      )
+
+      cdmSpy.returnValues[0].then(() => {
```

Comment:
If you set `onErrorSpy` to call `done()` after it runs then you don't need this.

Reply:
Or rather, you could replace `cmdSpy.returnValues[0].then(…)` with `onError.callsFake(…)` to run your expectations and call `done()` when the callback is called.

Reply:
I've made the lifecycles async and it makes sense to test that to me, which using the spies on them indirectly does.

Reply:
Fair enough, but this makes the LCV behave differently from viewers for images, audio and video, which won’t have this async `componentDidMount` behaviour.

Reply:
I think this will be true for the data visualization viewers.

Reply:
Text files need custom code to be fetched from the server too. I think it's still possible to write a set of generic tests for the viewer container props, without expecting a specific implementation in the container code. That can be addressed in #811.